### PR TITLE
localize decimal separator in video miniatures

### PR DESCRIPTION
**Pull request by kontrollanten**
## Description
[Some countries](https://en.wikipedia.org/wiki/Decimal_separator#Countries_using_decimal_comma) are using comma as decimal separator.
https://caniuse.com/internationalization

## Has this been tested?

- [x] 👍 yes, light tests as follows are enough

1) Switch language to Swedish.
1) See that decimal points for video views are comma.

## Screenshots

image

### CODE REVIEW

File: client/src/app/shared/shared-main/shared-main.module.ts [hunk 1]
```diff
@@ -150,6 +150,8 @@ import { VideoChannelService } from './video-channel'
   providers: [
     DatePipe,
 
+    DecimalPipe,
```
Comment (Chocobozzz) [inline client/src/app/shared/shared-main/shared-main.module.ts:153]: Where is used?

**Review by Chocobozzz — COMMENTED** [review]
Using https://angular.io/api/common/formatNumber would be simpler?

File: client/src/app/shared/shared-main/shared-main.module.ts (hunk 1 above)
Comment (kontrollanten) [reply client/src/app/shared/shared-main/shared-main.module.ts:153]: Oh, sorry, that's a relic.

**Comment by kontrollanten** [comment]
One thing that just hit me. Shall I put the Intl call in a try/catch, so it'll be handled gently in unsupported browsers?

File: client/src/app/shared/shared-main/angular/number-formatter.pipe.ts [hunk 2]
```diff
@@ -21,14 +16,25 @@ export class NumberFormatterPipe implements PipeTransform {
...
+    { max: 1000000000, type: 'M' }
+  ]
+  constructor (@Inject(LOCALE_ID) private localeId: string) {}
+
   transform (value: number) {
     const format = this.dictionary.find(d => value < d.max) || this.dictionary[this.dictionary.length - 1]
     const calc = value / (format.max / 1000)
     const integralPart = Math.floor(calc)
     const decimalPart = NumberFormatterPipe.getDecimalForNumber(calc)
+    const decimalSeparator = Intl.NumberFormat(this.localeId)
```
Comment (kontrollanten) [inline client/src/app/shared/shared-main/angular/number-formatter.pipe.ts:31]: Shall I put this in a try/catch, so it'll be handled gently in unsupported browsers?

**Pull request merged**